Modernize the Jumpbox Genesis Kit

jhunt opened the pull request:
This PR modernizes the Jumpbox Genesis Kit so that it handles things a
bit more ergonomically, using new features available as of Genesis 2.6.0
and above.

This has been vetted manually in both vSphere and AWS environments.

davidlohle reviewed (approved):
LGTM! Tested on GCP and vSphere, docs are updated and manuals look good.

patjones reviewed (commented):
Vetted on gcp and vsphere, docs look good. 👍

jhunt committed:
Modernize the Jumpbox Kit

This commit modernizes the Jumpbox kit to work well with Genesis 2.6
features and functionalities.  Primarily, this involves a heavier
emphasis on kit-provided hooks to do heavy lifting.

- Genesis 2.6.0 is the minimum version supported by this kit now.

- The `info` hook provides information about accessing the jumpbox.

- The `new` and `blueprint` hooks replace a lot of stuff that was in
  kit.yml (params and friends), while removing the complexity of the
  subkit decomposition.

- The `addon` hook provides for kit-specific functionality like SSHing
  to the jumpbox, managing VPN certificates, etc.

- The `post-deploy` hook gives the operator a gentle nudge in the
  direction of _next steps_.

jhunt commented:
I squash-merged offline, so that I could do the rebase in GH.

dennisjbell commented on MANUAL.md:
```diff
@@ -0,0 +1,194 @@
+  - `shell` - Login shell, i.e. `/bin/bash` or `/bin/zsh`
+  - `ssh_keys` - A list of public SSH keys to authorize for
+    password-less authentication against this account.
+
```
What happened to `env`?

jhunt replied:
It's being phased out in favor of forthcoming docker image changes.  It's valid, but undocumented, on purpose.

dennisjbell replied:
Premise accepted.

dennisjbell commented on MANUAL.md:
```diff
@@ -0,0 +1,194 @@
+
+# Available Features
+
+- `azure` - Replaces BOSH's native "availability zones"
```
azure should not be a feature, instead it should be detected in new, blueprint and check hooks

jhunt replied:
azure is not even a thing, auto-detected or otherwise.  we don't HA the jumpboxen.

dennisjbell commented on MANUAL.md:
```diff
@@ -0,0 +1,194 @@
+    example, a key has been lost or compromised.  The old
+    certificate will be revoked.
+
+  - `inventory` - Run the inventory errand against the deployment.
```
`certs` is missing from here, but is a viable option in the list

dennisjbell commented on hooks/addon:
```diff
@@ -0,0 +1,167 @@
+  echo "                users key in the first place.  The old certificate"
+  echo "                will be revoked."
+  echo
+  echo "  inventory     Take an inventory of software installed on the jumpbox"
```
certs and reissue-all are not listed in list, but are in the case statement

dennisjbell commented on hooks/blueprint:
```diff
@@ -0,0 +1,31 @@
+    ;;
+
+  proxy|azure)
+    echo >&2 "You no longer need to explicitly specify the 'proxy' feature."
```
replace 'proxy' with $want

jhunt replied:
Remove `azure` altogether; it's not a thing.

dennisjbell commented on hooks/info:
```diff
@@ -0,0 +1,5 @@
+set -eu
+ip=$(bosh vms --json | jq -r '.Tables[0].Rows[0].ips')
+
+describe "jumpbox ip: #C{$ip}"
```
Not needed, but a list of users for a given environment would probably be a good idea.

jhunt replied:
I believe the idea here was to remove the static_ips bits outright, and let the jumpboxen be dynamically IP'ed

dennisjbell replied:
I was referring to the user list no being needed, not the jumpbox ip, and its definitely not a blocker to approval.

dennisjbell reviewed (commented):
hooks/new: cidr2mask doesn't check against negative numbers, but that's probably not crucial.  Also, potential name conflict of addons (addons already mean something in genesis) under manifest for openvpn feature, why not just call it manifest/features/openvpn.yml

dennisjbell committed:
Updated as per review, added release notes, removed azure.

jhunt commented on hooks/new:
```diff
@@ -42,8 +42,9 @@ trap "rm -f $GENESIS_ROOT/.$GENESIS_ENVIRONMENT.yml" INT QUIT TERM EXIT
 echo "  name:    $GENESIS_KIT_NAME"
 echo "  version: $GENESIS_KIT_VERSION"
+echo "  features:"
+echo "    - (( replace ))"
```
This isn't needed.

jhunt committed:
Only print features list if we have features (hooks/new)
